addons/pos_self_order/models/pos_config.py

```diff
@@ -272,7 +272,7 @@ def load_self_data(self):
             }
         }
         response['pos.config']['data'][0]['_self_ordering_image_home_ids'] = self._get_self_ordering_attachment(self.self_ordering_image_home_ids)
-        response['pos.config']['data'][0]['_pos_special_products_ids'] = self._get_special_products().ids
+        response['pos.config']['data'][0]['_pos_special_products_ids'] = self._filter_special_products().ids
```

Comment:
Based on the logic that excludes the product, it's not needed to filter it here.